tiiuae/ghaf
# Hardware definition refactoring

## Description of changes

This PR introduces

1. Refactoring of hardware definition:
	- Moving hardware from common into a dedicated flake-module. This implies that hardware definition is no longer part of the VM namespace by default. 
	- Re-factoring `generation` parameter, which is now an option
2. Moving TPM related code to generic-x86_64 as hardware dependent module
3. Enabling fingerprint reader in the `guivm`




## Checklist for things done






## Testing


Tested by author:

1. Build + run `.#lenovo-x1-carbon-gen11-debug`
2. Builld + run `.#vm-debug`
3. Build + run `.#generic-x86_64-debug` (on x1)
4. Verified TPM stack running on x1 host
5. Fingerprint: (more info in commit message)
	- tested enrollment (+multiple fingers)
	- tested verification: sudo, systemd, and swaylock

Test instructions:
1. Normal build + run on all x86-based targets with functionality test
2. TPM: Verify that abrmd service runs on host
3. Fingerprint:
    - In terminal, use `fprintd-enroll` to enroll fingerprint. Run a `sudo` command and `systemctl restart systemd-timesyncd.service`, authenticate with fingerprint. In-depth testing not required at this point, as features are missing (see commit message). Expected outcome: fingerprint enrollment works, sudo and systemctl (system) commands can be authenticated with fingerprint
    - To test swaylock, enable `config.ghaf.graphics.labwc.lock.enable` and build, enroll fingerprint, then run `swaylock` (e.g., run in terminal), press Enter and use fingerprint reader to authenticate. Expected outcome: fingerprint authentication unlocks swaylock
